assets/js/modules/analytics/datastore/service.js

```diff
@@ -50,22 +50,28 @@ export const selectors = {
 	 */
 	getServiceURL: createRegistrySelector(
 		( select ) => ( state, { path, query } = {} ) => {
-			const userEmail = select( CORE_USER ).getEmail();
+			const baseURI = 'https://analytics.google.com/analytics/web/';
 
-			if ( userEmail === undefined ) {
-				return undefined;
+			let appendedURL = baseURI;
+
+			if ( query ) {
+				appendedURL = addQueryArgs( baseURI, query );
 			}
```

Comment:
Given we only ever use `baseURI` to define `appendedURL` and then later to redefine that variable entirely, I think it isn't really needed as a variable here. If we changed `appendedURL` to something more descriptive like `serviceURL` we could get rid of `baseURI` and the rest of the code might be a bit more descriptive, as we'd augment `serviceURL` with any query params, if present. Eg.:

```js
let serviceURL = 'https://analytics.google.com/analytics/web/';

if ( query ) {
	serviceURL = addQueryArgs(serviceURL, query);
}

if ( path ) {
	const sanitizedPath = `/${ path.replace( /^\//, '' ) }`;
	serviceURL = `${serviceURL}#${sanitizedPath}`;
}

const accountChooserServiceURL = select(CORE_USER).getAccountChooserURL(serviceURL);
```

(Also, we're mixing and matching URI and URL… we usually just use "URL" in our code… I know they're technically different, but it's probably easier to just stick with `URL`, especially as it's part of the selector's name 😅. I can see there's existing inconsistency in our codebase though, so it's not a big deal, especially re: `baseURI`. 🤷🏻‍♂️  )